com.microsoft.mrtk.core/Utilities/Attributes/VariableRangeAttribute.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit
+{
+    /// <summary>
+    /// An attribute that allows a particular field to be constrained to a range dependent on other variables
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Field)]
+    public sealed class VariableRangeAttribute : PropertyAttribute
```

Comment:
Do we want to make this `internal`, or do we think it's safe/useful enough for general use?

Reply:
I'll make it as internal to prevent accidentally taking on more maintenance work :)